saddle-registry.md

```diff
@@ -0,0 +1,201 @@
+# Saddle Registry
+
+The Saddle Registry is used for storage and retrieval of Saddle registries and contracts.
+
+## Master Registry
+
+| Network | Address                                                                                                            |
+| :------------ | :------------------------------------------------------------------------------------------------------------------------- |
+| `Ethereum`   | [0xc5ad17b98D7fe73B6dD3b0df5b3040457E68C045](https://etherscan.io/address/0xc5ad17b98D7fe73B6dD3b0df5b3040457E68C045#code) |
+
+The Master Registry can be interacted with the following functions:
+
+### Master Registry Contract Functions
+
+The Master Registry can be interacted with the following functions:
+
+* `resolveNameToLatestAddress`: Resolves a name input to the latest registry address.
+    ```
+        >>> await registry.resolveNameToLatestAddress(formatBytes32String('USDPool'))
+        '0xFb4DE84c4375d7c8577327153dE88f58F69EeC81'
+    ```
+* `resolveNameAndVersionToAddress`: Allows you to provide both a name and a specific version of the registry address.
+    ```
+        >>> await registry.resolveNameAndVersionToAddress(formatBytes32String('USDPool'), 0)
+        '0xFb4DE84c4375d7c8577327153dE88f58F69EeC81'
+    ```
+* `resolveNameToAllAddresses`: Returns an array of all addresses held in the registry of a given name.
+    ```
+        >>> await registry.resolveNameToAllAddresses(formatBytes32String('USDPool'))
+        ['0x0C8BAe14c9f9BF2c953997C881BEfaC7729FD314', '0xFb4DE84c4375d7c8577327153dE88f58F69EeC81']
+    ```
+* `resolveAddressToRegistryData`: Returns the registry input entry data inlcuding the name, version, and a boolean value of whether the provided address is the latest version.
+    ```
+    >>> const reg_data = await registry.resolveAddressToRegistryData('0xFb4DE84c4375d7c8577327153dE88f58F69EeC81')
+    >>> const name = parseBytes32String(reg_data['name'])
+    >>> const version = reg_data['version'].toNumber()
+    >>> const isLatest = reg_data['isLatest']
+    >>> console.log('name: ', name, ' version: ', version, ' isLatest: ', isLatest)
+    name:  USDPool  version:  1  isLatest:  true
+    ```
+
+## Pool Registry
+
+This contract holds the funtions to get deployment addresses and related info for all Saddle Pools
+
+| Network | Address                                                                                                            |
+| :------------ | :------------------------------------------------------------------------------------------------------------------------- |
+| `Ethereum`   | [0xFb4DE84c4375d7c8577327153dE88f58F69EeC81](https://etherscan.io/address/0xFb4DE84c4375d7c8577327153dE88f58F69EeC81#code) |
+
+### Pool Registry Public Variables:
+
+* `poolsIndexOfPlusOne(address poolAddress)`: Resolves an address input to the current pool vitual price.
+    ```python
```

Comment:
are we using JS or Python in our docs? Doesn't matter which, but it should only be one